week1/scary/kirikaza/Scary.hs

```diff
@@ -0,0 +1,19 @@
+module Main where
+
+import Data.Char (isAsciiLower, isAsciiUpper, ord)
+
+main :: IO ()
+main = do
+  file <- getContents
+  let words = lines file
```

Comment:
A useful habit is avoiding shadowing names (e.g. `words` and `sum` are defined in Prelude) because if you don't, one day it might bite you. Well, it won't with `-Wall` because the compiler will complain, but otherwise it will.

An example of where it can bite you:

```
main = do
  ...
  let sum = ...
  ...
  [you try to use the original 'sum' here and the compiler
  throws an ugly error and you spend a lot of time debugging it]
```

A worse example:

```
main = do
  ...
  let sum f = ...
  ...
  [you intend to use original 'sum' and types of original and redefined 'sum'
  match and the compiler doesn't complain and you have broken code]
```